packages/graphiql-react/src/editor/hooks.ts

```diff
@@ -204,14 +204,23 @@ export function useMergeQuery({ caller }: UseMergeQueryArgs = {}) {
   }, [queryEditor, schema]);
 }
 
-type UsePrettifyEditorsArgs = {
+export type UsePrettifyEditorsArgs = {
   /**
    * This is only meant to be used internally in `@graphiql/react`.
    */
   caller?: Function;
+  /**
+   * Invoked when the prettify callback is invoked
+   * @param query The current value of the query editor.
+   * @returns {string} The formatted query
+   */
+  onPrettifyQuery?: (query: string) => string;
```

Comment:
😎

Reply:
@dimaMachina along the lines of our previous discussion, i think we should discontinue exposing handlers and state via props, and expose them via hooks instead, then people can provide formatters as plugins. in this case it's simply a callback and no state, but for something like this that is more likely to be used by plugins, we should expose that instead, and provider a prettier plugin

Reply:
we already expose the use editor hooks and users can use the refs, so a compositional `@graphiql/react` experience already exists (as of course GraphiQL has been!), so we should just provide a simple example in `examples` of a custom editor?